Plugins/Tweaks/Tweaks/SneakAttackCritImmunity.cpp

```diff
@@ -223,58 +225,60 @@ void SneakAttackCritImmunity::CNWSCreature__ResolveDeathAttack_hook(CNWSCreature
+    else if (pThis->GetFlanked(pTarget))
     {
-        const uint8_t uncannyClasses[] =
-        {
-            Constants::ClassType::Barbarian,
-            Constants::ClassType::Rogue,
-            Constants::ClassType::Assassin,
-            Constants::ClassType::Shadowdancer
-        };
-        int attackerLevels = 0, defenderLevels = 0;
+        isDeathAttack = true;
 
-        for (uint8_t i = 0; i < 3; i++)
+        if (pTarget->m_pStats->HasFeat(Constants::Feat::UncannyDodge2))
         {
-            uint8_t attackerClass = pThis->m_pStats->GetClass(i);
-            uint8_t defenderClass = pTarget->m_pStats->GetClass(i);
-            for (size_t j = 0; j < sizeof(uncannyClasses)/sizeof(uncannyClasses[0]); j++)
+            const uint8_t uncannyClasses[] =
             {
-                attackerLevels += (attackerClass == uncannyClasses[j]) ? pThis->m_pStats->GetClassLevel(i, false) : 0;
-                defenderLevels += (defenderClass == uncannyClasses[j]) ? pTarget->m_pStats->GetClassLevel(i, false) : 0;
-            }
-        }
-        if (attackerLevels - defenderLevels >= Globals::Rules()->GetRulesetIntEntry("FLANK_LEVEL_RANGE", 4))
-        {
-            if (pThis->GetFlanked(pTarget)) // Bad function name, but this does the correct check
+                Constants::ClassType::Barbarian,
+                Constants::ClassType::Rogue,
+                Constants::ClassType::Assassin,
+                Constants::ClassType::Shadowdancer
+            };
+            int attackerLevels = 0, defenderLevels = 0;
+
+            for (uint8_t i = 0; i < 3; i++)
             {
-                isDeathAttack = (!pAttackData->m_bRangedAttack || fDistance < SNEAK_ATTACK_DISTANCE);
+                uint8_t attackerClass = pThis->m_pStats->GetClass(i);
+                uint8_t defenderClass = pTarget->m_pStats->GetClass(i);
+                for (size_t j = 0; j < sizeof(uncannyClasses) / sizeof(uncannyClasses[0]); j++)
+                {
+                    attackerLevels += (attackerClass == uncannyClasses[j]) ? pThis->m_pStats->GetClassLevel(i, false) : 0;
+                    defenderLevels += (defenderClass == uncannyClasses[j]) ? pTarget->m_pStats->GetClassLevel(i, false) : 0;
+                }
             }
+
+            isDeathAttack = attackerLevels - defenderLevels >= Globals::Rules()->GetRulesetIntEntry("FLANK_LEVEL_RANGE", 4);
         }
     }
 
     if (isDeathAttack)
     {
         if (pTarget->m_pStats->GetEffectImmunity(Constants::ImmunityType::SneakAttack, pThis, true))
         {
-            CNWCCMessageData *pData = new CNWCCMessageData;
+            CNWCCMessageData* pData = new CNWCCMessageData;
             pData->SetObjectID(0, pTarget->m_idSelf);
             pData->SetInteger(0, 134);
             pAttackData->m_alstPendingFeedback.Add(pData);
         }
         else
         {
-            pAttackData->m_bDeathAttack = 1;
+            pAttackData->m_bSneakAttack = 1;
```

Comment:
Copy/paste error?

Reply:
Yeah, thanks for catching it. Kinda pushed that commit quickly so I wouldn't forget about the death attack.